nixos/modules/services/networking/sing-box.nix

```diff
@@ -75,6 +96,13 @@ in
       };
       wantedBy = [ "multi-user.target" ];
     };
-  };
 
+    users = {
+      users.${cfg.user} = {
+        isSystemUser = true;
+        inherit (cfg) group;
+      };
+      groups.${cfg.group} = { };
+    };
+  };
```

Comment:
Normally we only automatically create the user/group when it's not changed from the default, e.g.
```
nixos/modules/services/mail/postfix.nix
924:        users.users = lib.optionalAttrs (user == "postfix") {
```